src/App.tsx

```diff
@@ -17,18 +62,30 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                status={status}
+                query={query}
+                onQueryChange={setQuery}
+                onStatusChange={setStatus}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!areTodosLoaded && <Loader />}
+
+              <TodoList
+                todos={visibleTodos}
+                selectedTodo={selectedTodo}
+                onTodoSelect={setSelectedTodo}
+              />
```

Comment:
```suggestion
              {!areTodosLoaded && <Loader />}

              {areTodosLoaded && <TodoList
                todos={visibleTodos}
                selectedTodo={selectedTodo}
                onTodoSelect={setSelectedTodo}
              />}
```

I'm not sure, but maybe you should try this because your Loader and TodoLis render at the same time. Look on screens. The second screen I take from the working page. But... it's just advice
![image](https://user-images.githubusercontent.com/87998640/236395851-cbebf98f-fe80-4f2f-b0e1-de54dcffbcee.png)
![image](https://user-images.githubusercontent.com/87998640/236396373-26cedf88-fa2d-4c14-af4e-4a6041ebc321.png)